tests/unit/hosts/test_views.py

```diff
@@ -573,3 +573,135 @@ def test_when_requesting_user_is_not_part_of_team(self):
+            },
+        )
+
+    def test_challenge_host_does_not_exist_in_team(self):
+        self.url = reverse_lazy(
+            "hosts:delete_challenge_host_from_team",
+            kwargs={
+                "challenge_host_team_pk": self.challenge_host_team.pk,
+                "challenge_host_pk": self.challenge_host2.pk + 1,
+            },
+        )
+
+        expected = {"error": "ChallengeHost does not exist"}
+
+        response = self.client.delete(self.url, {})
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_406_NOT_ACCEPTABLE)
+
+    def test_when_challenge_host_team_does_not_exist(self):
+        self.url = reverse_lazy(
+            "hosts:delete_challenge_host_from_team",
+            kwargs={
+                "challenge_host_team_pk": self.challenge_host_team.pk + 1,
+                "challenge_host_pk": self.challenge_host2.pk,
+            },
+        )
+
+        expected = {"error": "ChallengeHostTeam does not exist"}
+
+        response = self.client.delete(self.url, {})
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_406_NOT_ACCEPTABLE)
+
+    def test_when_challenge_host_is_admin_and_wants_to_delete_himself(self):
+        self.url = reverse_lazy(
+            "hosts:delete_challenge_host_from_team",
+            kwargs={
+                "challenge_host_team_pk": self.challenge_host_team.pk,
+                "challenge_host_pk": self.challenge_host.pk,
+            },
+        )
+
+        expected = {
+            "error": "You are not allowed to remove yourself since you are admin. Please delete the team if you want to do so!"  # noqa: ignore=E501
+        }
+
+        response = self.client.delete(self.url, {})
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_406_NOT_ACCEPTABLE)
+
+    def test_when_challenge_host_does_not_have_permissions_to_remove_another_challenge_host(
+        self
+    ):
+        self.url = reverse_lazy(
+            "hosts:delete_challenge_host_from_team",
+            kwargs={
+                "challenge_host_team_pk": self.challenge_host_team.pk,
+                "challenge_host_pk": self.challenge_host2.pk,
+            },
+        )
+
+        self.user3 = User.objects.create(
+            username="user3",
+            email="user3@platform.com",
+            password="user3_password",
+        )
+
+        EmailAddress.objects.create(
+            user=self.user3,
+            email="user3@platform.com",
+            primary=True,
+            verified=True,
+        )
+
+        self.challenge_host3 = ChallengeHost.objects.create(
+            user=self.user3,
+            team_name=self.challenge_host_team,
+            status=ChallengeHost.ACCEPTED,
+            permissions=ChallengeHost.ACCEPTED,
+        )
+
+        self.client.force_authenticate(user=self.user3)
```

Comment:
Can we define all this in the `setUp`?

Reply:
> Can we define all this in the `setUp`?

@nikochiko in one test class only one setUp is supported and if i put all these in the main setUp the particular function is unable to access it and throwing error.

Reply:
@Kajol-Kumari
>
> @nikochiko if i put all these in the main setUp the particular function is unable to access it and throwing error.

I didn't understand what you mean. 
I am saying we could define it in [this](https://github.com/Cloud-CV/EvalAI/pull/2660/files#diff-622550d062dfae0cf9c89d9c3b1ab078R579) `setUp` method like we did for other functions. Please let me know if I am missing something.

Reply:
> @Kajol-Kumari
> 
> > @nikochiko if i put all these in the main setUp the particular function is unable to access it and throwing error.
> 
> I didn't understand what you mean.
> I am saying we could define it in [this](https://github.com/Cloud-CV/EvalAI/pull/2660/files#diff-622550d062dfae0cf9c89d9c3b1ab078R579) `setUp` method like we did for other functions. Please let me know if I am missing something.

I was doing the wrong way. Got my fault and fixed it. Also there were two test classes written pointing to the same url as that of "DeleteParticularChallengeHost". I have commented them along with providing a comment on the top.